internal/server/server.go

```diff
@@ -1234,3 +1234,24 @@ func (s Server) SeenSince(r *http.Request, timeAgo time.Duration) bool {
 func (s Server) IsEmail(val string) bool {
 	return s.emailRe.MatchString(val)
 }
+
+func (s Server) GetSignOnAttempts(email string) int {
+	numberOfAttempts := 0
+	attempts, found := s.CacheGet(email)
+	if found {
+		numberOfAttempts, err := strconv.Atoi(string(attempts))
+		if err == nil {
+			return numberOfAttempts
+		}
+	}
+
+	return numberOfAttempts
+}
+
+func (s Server) SaveSignOnAttempts(email string) {
```

Comment:
please dont create custom functions inside server.go, add the logic in the handler and use the existing functions SetCache/GetCache. Otherwise this file becomes too large

Reply:
sure.